addons/sale/controllers/portal.py

```diff
@@ -189,7 +189,7 @@ def portal_order_page(self, order_id, report_type=None, access_token=None, messa
             session_obj_date = request.session.get('view_quote_%s' % order_sudo.id)
             if session_obj_date != now and request.env.user.share and access_token:
                 request.session['view_quote_%s' % order_sudo.id] = now
-                body = _('Quotation viewed by customer %s', order_sudo.partner_id.name)
+                body = _('Quotation viewed by customer %s', request.env.user.partner_id.name)
```

Comment:
Well, if the quotation is viewed anonymously (with the access token), we do not want to log "Quotation viewed by customer Public User" either, that's why we were using the SO partner and not the current user name.

![image](https://github.com/odoo/odoo/assets/19364378/0f1a2031-59d3-4088-869f-0118c8ca546e)

Reply:
Should we check for that condition to fallback if the quotation is viewed by anonymously?

Reply:
I have revised the commit to make sure to use the right partner name except when the quotation is viewed anonymously. Please help review.